### PR TITLE
PLATOPS-1585: Add upstream trigger

### PR DESCRIPTION
**Pull request by dabd**
Based on ReverseBuildTrigger, this trigger enables building a job based
on the successful completion of upstream jobs.

### CODE REVIEW

File: src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        this.condition = condition
+    }
+
+    /**
+     * Creates a {@link jenkins.triggers.ReverseBuildTrigger} based on the successful completion of any of a list of upstream jobs.
+     * @param upstreamJobs String with a comma separated list of jobs
+     * @param {@link uk.gov.hmrc.jenkinsjobbuilders.domain.Condition} one of SUCCESS, UNSTABLE, FAILURE
+     * @return UpstreamTrigger
+     */
+    static UpstreamTrigger upstreamTrigger(String upstreamJobs, Condition condition) {
```
Comment (konradwudkowski) [inline src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy]: Should users define `condidtion` always? They will probably usually want success anyway. Maybe `condiditon` could be defaulted on line 7 (`Condition condition = Condition.SUCCESS`) and we provide a `withCondition(Condidtion condition)` where people could override in rare cases?

**Comment by konradwudkowski** [comment]
Readme of this Library has a change log where we added every modification, let's continue this.

File: src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy (hunk 1 above)
Comment (dabd) [reply src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy]: Thanks, we'll make those changes.

Comment (dabd) [reply src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/trigger/UpstreamTrigger.groovy]: Comments addressed by https://github.com/hmrc/jenkins-job-builders/pull/41/commits/3f156f857ce5ab2bcf0d82008ad4b04287805c9f

**Pull request merged**